redhat-director-scripts/common/install_puppet_module.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash -x
+
+set -e
+
+if [ $# -lt 1 ];then
+   echo "Script takes exacyly 1 argument"
+   echo -e "./install_puppet_module.sh <overcloud_full_image_path>"
+   echo -e "Example"
+   echo -e "./install_puppet_module.sh /home/stack/images/overcloud-full.qcow2"
+   exit 1
+fi
+
+overcloud_full_image_path=$1
+tmp_working_dir="/tmp/triliovault"
+overcloud_full_image_name=`basename $overcloud_full_image_path`
+
+
+rm -rf $tmp_working_dir
+mkdir -p $tmp_working_dir
+cp $overcloud_full_image_path ${tmp_working_dir}/
+cp trilio.repo ${tmp_working_dir}/
+
+cd ${tmp_working_dir}/
+virt-customize --selinux-relabel -a ${overcloud_full_image_name} --upload trilio.repo:/etc/yum.repos.d/
+virt-customize --selinux-relabel -a ${overcloud_full_image_name} --install puppet-triliovault
+virt-customize --selinux-relabel -a ${overcloud_full_image_name} --run-command  'ln -s /usr/share/openstack-puppet/modules/trilio /etc/puppet/modules/trilio && rm -f /etc/yum.repos.d/trilio.repo'
```

Comment:
I think you can do all three of these in a single execution. Additionally you need to clear the machine-id at the end (it causes problems with computes)

virt-sysprep --operation machine-id --format qcow2 -a /image-build/overcloud-full.qcow2

You may also need to export LIBGUESTFS_BACKEND=direct

Reply:
Thank you @mwhahaha . I will implement these comments.

Reply:
@mwhahaha  Implemented all suggestions. Please have a look whenever you get time.

Reply:
@mwhahaha I tried running all commands using single file using option "--commands-from-file" but, it's not waiting for previous command excution to complete before starting next command execution.
That's why, rm -f and ln -s commands are failing.

I kept those commands separate and clubbed other commands in one file.